### PR TITLE
Pull Request for Issue2343: Modifying limits functionality in BioXASControlEditor

### PR DESCRIPTION
**Pull request by helfrij**
Changes:
- Created new class BioXASValueSetpointEditor that's a combo spinbox/combobox as appropriate, depending on the InputType. The class also has an InputStatus property that's used to style the widget according to the whether or not the input is okay.
- Created new stylesheet for the setpoint editor that gives it a red border for bad input values.
- Modified AMAppController to apply stylesheets while setting up the user interface, instead of in the constructor. This allows for app controller subclasses to reimplement applyStylesheets() properly.
- Created new class BioXASValueSetpointEditorDialog, a dialog with the new setpoint editor and OK/Cancel buttons.
- Modified BioXASValueEditor to use the new dialog instead of the old default 'QInputDialog' option.
- Modified BioXASSSRLMonochromatorRegionControlEditor to inherit from BioXASControlEditor instead of AMExtendedControlEditor.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you review?

File: source/application/BioXAS/BioXASAppController.cpp [hunk 1]
```diff
@@ -67,6 +67,45 @@ void BioXASAppController::shutdown()
...
+	QFile qss1(":/AMToolButton.qss");
+
+	if (qss1.open(QFile::ReadOnly))
+		stylesheet.append(QString("\n\n%1").arg(QLatin1String(qss1.readAll())));
+
+	qss1.close();
+
+	// AMDeadTimeButton
+
+	QFile qss2(":/AMDeadTimeButton.qss");
```
Comment (XuetaoLiu-CLS) [inline source/application/BioXAS/BioXASAppController.cpp]: shouldn't we put those stylesheet into super class?

File: source/ui/BioXAS/BioXASPersistentView.cpp [hunk 2]
```diff
@@ -28,6 +28,7 @@ along with Acquaman.  If not, see <http://www.gnu.org/licenses/>.
 #include "ui/BioXAS/BioXASCryostatView.h"
 #include "ui/BioXAS/BioXASSIS3820ScalerChannelsView.h"
 #include "ui/AMToolButton.h"
+#include "ui/BioXAS/BioXASValueSetpointEditor.h"
```
Comment (XuetaoLiu-CLS) [inline source/ui/BioXAS/BioXASPersistentView.cpp]: is this used anywhere in the BioXASPersistentView?

**Comment by XuetaoLiu-CLS** [comment]
Looks good. other than a question.

Time Estimate: 10 minutes

File: source/application/BioXAS/BioXASAppController.cpp (hunk 1 above)
Comment (helfrij) [inline source/application/BioXAS/BioXASAppController.cpp]: Unfortunately, the stylesheet needs to be applied all at once at the AppController level; subsequent calls to `qApp->setStylesheet(...)` overwrite previous entries. We could do something like a general `AMDatamanAppController::getStylesheet(...)` that could be reimplemented for each application as desired--that's an idea.

File: source/ui/BioXAS/BioXASPersistentView.cpp (hunk 2 above)
Comment (helfrij) [inline source/ui/BioXAS/BioXASPersistentView.cpp]: It shouldn't be. Probably just leftover from testing. 😃

**Comment by dretrex** [comment]
This looks really good!
Time Estimate: 10 minutes

**Pull request merged**